pkg/fakerp/dns.go

```diff
@@ -27,23 +28,66 @@ func newDNSManager(ctx context.Context, log *logrus.Entry, subscriptionID, dnsRe
 	}
 
 	return &dnsManager{
-		zc:               dns.NewZonesClient(ctx, log, subscriptionID, authorizer),
-		rsc:              dns.NewRecordSetsClient(ctx, log, subscriptionID, authorizer),
+		zonesClient:      dns.NewZonesClient(ctx, log, subscriptionID, authorizer),
+		recordSetsClient: dns.NewRecordSetsClient(ctx, log, subscriptionID, authorizer),
 		dnsResourceGroup: dnsResourceGroup,
 		dnsDomain:        dnsDomain,
+		log:              log,
 	}, nil
 }
 
-func (dm *dnsManager) createOrUpdateZone(ctx context.Context, resourceGroup, zoneName, parentResourceGroup, parentZoneName string) error {
-	zone, err := dm.zc.CreateOrUpdate(ctx, resourceGroup, zoneName, azdns.Zone{
+func (dm *dnsManager) createOrUpdateAroDns(ctx context.Context, cs *api.OpenShiftManagedCluster) error {
```

Comment:
Does this have to be part of `DNSmanager`. Looks like product specific. What if we move it to outside `utils` package:
`https://github.com/openshift/openshift-azure/blob/c010f7798a1c33b2507fbe1acc4df33cbf4a827b/pkg/fakerp/fakerp.go#L172` and https://github.com/openshift/openshift-azure/blob/e7b4e323e1f7da47aaf770ece87c218fff584e76/pkg/fakerp/customer_handlers.go#L33 becomes one call to function in utils, which initializes DNS manager and does all the ARO/OCP logic. This way we keep DNS manager product agnostic? WDYT?

Reply:
This project is about one product `ARO` and the way things were setup seems fine to me. I don't feel strongly against any further refactorings however that can be done in a follow up.

Reply:
it true. but thinking from code perspective - reusable block always. More code can be re-used without ties to the product - better. Like nothing would stop you to just pull out DNS manager to separate git repository as a separate standalone package. But not a priority now :) just flow of ideas :)